### PR TITLE
Fixing Identity 4094

### PR DESCRIPTION
**Pull request by DMHP**

### CODE REVIEW

File: components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider_finish.jsp [hunk 1]
```diff
@@ -228,8 +228,9 @@
             client.removeServiceProvier(serviceProviderDTO.getIssuer());
         }
         status = client.addServiceProvider(serviceProviderDTO);
-        attributeConsumingServiceIndex = client.getServiceProvider(serviceProviderDTO.getIssuer()).getAttributeConsumingServiceIndex();
-
+        if((client.getServiceProvider(serviceProviderDTO.getIssuer())!=null)){
```
Comment (darshanasbg) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider_finish.jsp]: not properly formatted.

File: components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider_finish.jsp [hunk 2]
```diff
@@ -255,17 +256,29 @@
 
     if (applicationComponentFound) {
         if (status) {
+        if(attributeConsumingServiceIndex!=null){
```
Comment (darshanasbg) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider_finish.jsp]: format this block

File: components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/FileBasedConfigManager.java [hunk 3]
```diff
@@ -62,6 +63,14 @@ public static FileBasedConfigManager getInstance() {
     }
 
     /**
+     *Get the list of issuers which has been added through the file
+     * @return a String list of file based issuers
+     */
+    public static List<String> getIssuerList() {
```
Comment (darshanasbg) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/FileBasedConfigManager.java]: Do we need to introduce new method here? Can't we use
````java
SSOServiceProviderConfigManager.getServiceProvider(String issuerName)
````
for this?

File: components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider_finish.jsp (hunk 1 above)
Comment (DMHP) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider_finish.jsp]: Formatted

File: components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider_finish.jsp (hunk 2 above)
Comment (DMHP) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider_finish.jsp]: Formatted

File: components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/FileBasedConfigManager.java (hunk 3 above)
Comment (DMHP) [inline components/sso-saml/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/admin/FileBasedConfigManager.java]: Yes we can reuse the existing method. I have changed the code

**Pull request merged**